From 249e3399d3ed008abe249a384897632e993d98f6 Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Thu, 17 Mar 2022 11:43:06 +0200
Subject: [PATCH] [SW-0] gaudi2: ppboot: set the stack pointer at ppboot entry point

As of today, stack pointer of ppboot (__arc__) is not set explicitly
at the entry point. It uses stack pointer of ARC ROM for its execution.

When debugging ppboot via JTAG it was observed that SP was set to SRAM
address space instead of ARC1_DCCM as ARC_ROM was not run. To avoid
such, stack pointer is set explicitly at the entry point to __stack_top
which falls in ARC1_DCCM space.

Change-Id: I4f8efad43889fd59878cbd09d00a85ba5179b05e
---

diff --git a/app/ppboot/crt0_ppboot.S b/app/ppboot/crt0_ppboot.S
index c9dfb71..dc5c8d4 100644
--- a/app/ppboot/crt0_ppboot.S
+++ b/app/ppboot/crt0_ppboot.S
@@ -17,6 +17,9 @@
 .global	lf2flt
 lf2flt:
 
+	mov	sp, __stack_top
+	mov	fp, sp
+
 	/* Disable un-aligned checks on data memory */
 	lr r0, [STATUS32]
 	bset r0, r0, 0x13
